File: leetCode/Python2/191-numberOfOneBits.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        for i in xrange(32):
            count += (n & 0x01)
            n >>= 1
        return count

if __name__ == "__main__":
    s = Solution()
    print s.hammingWeight(11)
    print s.hammingWeight(3)
    print s.hammingWeight(4)
    print s.hammingWeight(0)

File: leetCode/C++/191-numberOfOneBits.cpp
#include <iostream>

using namespace std;

class Solution1 { // use the power of two property to solve it.
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
        while(n){
            count ++;
            n &= (n - 1);
        };
        return count;
    }
};

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
        for (int i = 31; i > -1; --i) {
            count += (n & 1);
            n >>= 1;
        }
        return count;
    }
};

int main(int argc, char *argv []){
    Solution s;
    cout << "count: " << s.hammingWeight(9) << endl;
    cout << "count: " << s.hammingWeight(7) << endl;
    return 0;
};